openc3-cosmos-cmd-tlm-api/app/controllers/storage_controller.rb

```diff
@@ -103,11 +103,18 @@ def get_download_presigned_request
     bucket_name = ENV[params[:bucket]] # Get the actual bucket name
     raise "Unknown bucket #{params[:bucket]}" unless bucket_name
     path = sanitize_path(params[:object_id])
-    bucket.check_object(bucket: bucket_name, key: path)
-    result = bucket.presigned_request(bucket: bucket_name,
-                                      key: path,
-                                      method: :get_object,
-                                      internal: params[:internal])
+    puts "get_download_presigned_request bucket:#{bucket_name} path:#{path}"
+    if bucket.check_object(bucket: bucket_name, key: path)
+      result = bucket.presigned_request(bucket: bucket_name,
+                                        key: path,
+                                        method: :get_object,
+                                        internal: params[:internal])
+    else
+      result = bucket.presigned_request(bucket: bucket_name,
```

Comment:
Should check if targets_modified even exists in the path before jumping to the second presigned_request call.